simulationdataschema/utils/utils.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD.
+# See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from nomad.datamodel.data import ArchiveSection
+
+
+def get_sub_section_from_section_parent(
+    section: ArchiveSection, sub_section: str, logger
+) -> ArchiveSection:
+    """
+    Gets the sub_section of a section by performing a seesaw move, going to the parent of
+    section, and then going down to sub_section. Example, if section is `Symmetry`, and we
+    want to resolve `AtomicCell` (sub_section), this methods goes up to `ModelSystem` from `Symmetry`,
+    and then goes down to `AtomicCell`.
+
+    If the sub_section is a list, it returns the first element of the list. If the sub_section is
+    a single section, it returns the section.
```

Comment:
Well, I would like to return one section. This is because otherwise we would have to deal with `isinstance` outside this method

Reply:
But is that generic enough? It sounds like for now you just want data from something that's representative, but this might also be used to apply normalization to all siblings, in which case you do want the `list`.

Don't forget that functions in `utils` are supposed to be very generic. Too much specialization for the sake of a bit of convenience will kill the function's utility.

Anyhow, I think that for now you can just add an option parameter and have it default to your preference.
We can revisit the function scope later on.

Reply:
> But is that generic enough? 

Yes

> It sounds like for now you just want data from something that's representative

No, I only want to retrieve one of the ArchiveSections. I can add another argument as the element of the list `i_sibling` which defaults to the first, 0.

Reply:
Good addition. You're almost there:

`index_sibling: Optional[int] = 0,` where `None` stands for the full list.

Tbh, this solution is still very ad-hoc, but with this last addition, you can manage (for now).
This is a symptom of us lacking a proper generic search functionality. Let me see with Area D, how we can extend `MSection` with my `Hierearchy` design.